backend/src/layer/data/image_validation/api_request.rs

```diff
@@ -0,0 +1,115 @@
+use crate::interface::image_validation::ImageValidationError::InvalidResponse;
+use crate::interface::image_validation::Result;
+use crate::layer::data::image_validation::json_structs::{SafeSearchJson, SafeSearchResponseJson};
+use google_jwt_auth::AuthConfig;
+use std::fs;
+
+static API_REST_URL: &str = "https://vision.googleapis.com/v1/images:annotate";
+static API_USAGE: &str = "https://www.googleapis.com/auth/cloud-vision";
+static PROJECT_ID_HEADER: &str = "x-goog-user-project";
+static REQUEST_TYPE: &str = "SAFE_SEARCH_DETECTION";
+static CONTENT_TYPE: &str = "application/json";
+static TOKEN_LIFETIME: i64 = 3600;
+static CHARSET: &str = "utf-8";
+
+/// The [`ApiRequest`] struct is used to send images and
+/// requests safe-search results from the api rest interface.
+pub struct ApiRequest {
+    google_project_id: String,
+    auth_config: AuthConfig,
+}
+
+impl ApiRequest {
+    /// This method is used to create a new instance of the [`ApiRequest`] struct.
+    /// # Params
+    /// `service_account_json_path`<br>
+    /// This param contains the json as string. The data inside the json is used to
+    /// establish a connection to the api interface and authenticate the client.<br>
+    /// `google_project_id`<br>
+    /// This id is needed to verify the client/caller of the request.
+    /// The `project_id` can be obtained in the google console.
+    /// # Errors
+    /// If json could not be read or the authentication struct could not be build, an error will be returned.
+    /// See [`crate::interface::image_validation::ImageValidationError`] for more info about the errors.
+    /// # Return
+    /// The mentioned [`ApiRequest`] struct.
+    pub fn new(service_account_json_path: String, google_project_id: String) -> Result<Self> {
+        let json_str = fs::read_to_string(service_account_json_path)?;
```

Comment:
i would do that in the config reader. (this should also be done async usink tokios variant of read_to_string)